mu-trees/addon/module-registries/requirejs.js

```diff
@@ -43,51 +43,55 @@ export default class RequireJSRegistry {
     return segments;
   }
 
-  _detectModule(specifierString, lookupMethod) {
-    let specifier = deserializeSpecifier(specifierString);
-
+  _detectModule(specifier, lookupDefault, lookupNamed) {
     let segments = this._baseSegments(specifier);
     let basePath = `${segments.join('/')}`;
     let typedPath = `${basePath}/${specifier.type}`;
 
-    let lookupResult = lookupMethod(typedPath);
+    let lookupResult = lookupDefault(typedPath);
 
-    if (
-      !lookupResult &&
-      this._config.collections[specifier.collection].defaultType === specifier.type
-    ) {
-      lookupResult = lookupMethod(basePath);
+    if (!lookupResult) {
+      if (this._checkDefaultType(specifier)) {
+        lookupResult = lookupDefault(basePath);
```

Comment:
@mixonic My inclination is to be conservative and say we should not for now. This is a very hot path - extra lookups at this level must be considered as a multiplier against all resolution checks. I'd prefer to be quite deliberate about adding this (if we ever do) because it will be much easier to add than take away later.